ts/Stock/Indicators/DMI/DMIIndicator.ts

```diff
@@ -0,0 +1,398 @@
+ *
+ * */
+
+'use strict';
+
+/* *
+ *
+ *  Imports
+ *
+ * */
+
+import type {
+    DMIOptions,
+    DMIParamsOptions
+} from './DMIOptions';
+import type DMIPoint from './DMIPoint';
+import type IndicatorValuesObject from '../IndicatorValuesObject';
+import type LineSeries from '../../../Series/Line/LineSeries';
+import MultipleLinesMixin from '../../../Mixins/MultipleLines.js';
+import SeriesRegistry from '../../../Core/Series/SeriesRegistry.js';
+const {
+    seriesTypes: {
+        sma: SMAIndicator
+    }
+} = SeriesRegistry;
+import U from '../../../Core/Utilities.js';
+const {
+    correctFloat,
+    extend,
+    isArray,
+    merge
+} = U;
+
+/* *
+ *
+ *  Class
+ *
+ * */
+
+/**
+ * The Directional Movement Index (DMI) series type.
+ *
+ * @private
+ * @class
+ * @name Highcharts.seriesTypes.dmi
+ *
+ * @augments Highcharts.Series
+ */
+class DMIIndicator extends SMAIndicator {
+    /**
+     * Directional Movement Index (DMI).
+     * This series requires the `linkedTo` option to be set and should
+     * be loaded after the `stock/indicators/indicators.js` file.
+     *
+     * @sample stock/indicators/dmi
+     *         DMI indicator
+     *
+     * @extends      plotOptions.sma
+     * @since        next
+     * @product      highstock
+     * @excluding    allAreas, colorAxis, joinBy, keys, navigatorOptions,
+     *               pointInterval, pointIntervalUnit, pointPlacement,
+     *               pointRange, pointStart, showInNavigator, stacking
+     * @requires     stock/indicators/indicators
+     * @requires     stock/indicators/dmi
+     * @optionparent plotOptions.dmi
+     */
+    public static defaultOptions: DMIOptions = merge(SMAIndicator.defaultOptions, {
+        /**
+         * @excluding index
+         */
+        params: {
+            index: void 0, // DMI has unchangeable index
+            period: 14
+        },
+        marker: {
+            enabled: false
+        },
+        tooltip: {
+            pointFormat: '<span style="color:{point.color}">\u25CF</span><b> {series.name}</b><br/>' +
+                '<span style="color:{point.color}">DX</span>: {point.y}<br/>' +
+                '<span style="color:{point.series.options.plusDILine.styles.lineColor}">+DI</span>' +
+                    ': {point.plusDI}<br/>' +
+                '<span style="color:{point.series.options.minusDILine.styles.lineColor}">-DI</span>' +
+                    ': {point.minusDI}<br/>'
+        },
+        /**
+         * +DI line options.
+         */
+        plusDILine: {
+            /**
+             * Styles for the +DI line.
+             */
+            styles: {
+                /**
+                 * Pixel width of the line.
+                 */
+                lineWidth: 1,
+                /**
+                 * Color of the line.
+                 *
+                 * @type {Highcharts.ColorString}
+                 */
+                lineColor: '#00ff00'
```

Comment:
Maybe we might use some predefined colors? Like it is done in PC indicator
`lineColor: palette.colors[2],` which is `#90ed7d` and at least for me this kind of green is more pleasant to the eye than `#00ff00`.
Just an idea to discuss

Reply:
Sure, I agree.